### PR TITLE
deployment: Add elasticsearch keystore commands

### LINKED ISSUE #213: Add support for endpoints to manage ES keystore
With the migration to /deployments endpoint a new endpoint will be added to allow users to list and update ES keystore.

The new endpoint /deployments/{deployment_id}/elasticsearch/{es-ref-id}/keystore will support GET and PATCH to list and create or update a keystore entry.

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds a new `elasticsearch` subcommand to the `deployment` command and a
`keystore` subcommand to the `deployment elasticsearch` command. Results
in:

* `deployment elasticsearch keystore show`
* `deployment elasticsearch keystore update|set`

## Related Issues

Resolves #213 

## How Has This Been Tested?

Unit and manually tested against an ESS deployment.

## Types of Changes

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cmd/deployment/elasticsearch/command.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"github.com/spf13/cobra"
+
+	cmdeskeystore "github.com/elastic/ecctl/cmd/deployment/elasticsearch/keystore"
+)
+
+// Command is the deployment subcommand
+var Command = &cobra.Command{
+	Use:     "elasticsearch",
+	Short:   "Manages a deployment's elasticsearch resources",
```
Comment (karencfv) [inline cmd/deployment/elasticsearch/command.go:29]: I remember @nrichers said something about not using `'s` if possible. How about "Manages Elasticsearch resources" or "Manages Elasticsearch resources from a deployment"

File: cmd/deployment/elasticsearch/keystore/command.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// under the License.
+
+package cmdeskeystore
+
+import "github.com/spf13/cobra"
+
+// Command is the deployment subcommand
+var Command = &cobra.Command{
+	Use:     "keystore",
+	Short:   "Manages a deployment's elasticsearch resource keystore",
```
Comment (karencfv) [inline cmd/deployment/elasticsearch/keystore/command.go:25]: same as above

File: cmd/deployment/elasticsearch/keystore/show.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	"github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi/eskeystoreapi"
+	sdkcmdutil "github.com/elastic/cloud-sdk-go/pkg/util/cmdutil"
+
+	"github.com/elastic/ecctl/pkg/ecctl"
+)
+
+var showCmd = &cobra.Command{
+	Use:     "show <deployment id> [--ref-id <ref-id>]",
+	Short:   "Shows an Elasticsearch keystore contents",
```
Comment (karencfv) [inline cmd/deployment/elasticsearch/keystore/show.go:31]:
```suggestion
	Short:   "Shows the settings from the Elasticsearch resource keystore",
```
copy pasted from the apidocs 😄

File: cmd/deployment/elasticsearch/keystore/update.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	sdkcmdutil "github.com/elastic/cloud-sdk-go/pkg/util/cmdutil"
+
+	"github.com/elastic/ecctl/pkg/ecctl"
+)
+
+const updateLong = `Changes the contents of the specified deployment's elasticsearch
+keystore by using the PATCH method, the payload is a partial payload where
+any ignored current keystore items, are not removed, unless the secrets are
+set to "null": {"secrets": {"my-secret": null}}.`
```
Comment (karencfv) [inline cmd/deployment/elasticsearch/keystore/update.go:37]:
```suggestion
const updateLong = `Changes the contents of the Elasticsearch resource keystore from the specified deployment by using the PATCH method. The payload is a partial payload where
any ignored current keystore items are not removed, unless the secrets are
set to "null": {"secrets": {"my-secret": null}}.`
```

File: cmd/deployment/elasticsearch/keystore/update.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		}
+
+		return ecctl.Get().Formatter.Format("deployment/eskeystore_show", res)
+	},
+}
+
+func init() {
+	Command.AddCommand(updateCmd)
+	updateCmd.Flags().String("ref-id", "", "Optional ref_id to use for the Elasticsearch resource, auto-discovered if not specified.")
+	updateCmd.Flags().String("file", "", "Required json formatted file path with the keystore secret contents")
```
Comment (karencfv) [inline cmd/deployment/elasticsearch/keystore/update.go:77]:
```suggestion
	updateCmd.Flags().StringP("file", "f", "", "Required json formatted file path with the keystore secret contents")
```

File: cmd/deployment/command.go [hunk 6]
```diff
@@ -41,6 +41,6 @@ func init() {
 		cmddeploymentnote.Command,
 		cmddeploymentplan.Command,
 		cmddeploymentresource.Command,
-		cmddeploymenttemplate.Command,
+		cmdelasticsearch.Command,
```
[the remark this answers is not in the document]
Comment (marclop) [reply cmd/deployment/command.go:44]: omg, what a silly mistake

**Comment by marclop** [comment]
Thanks Karen!

**Pull request merged**